### PR TITLE
Move isKind and isAnnotation to Type

### PR DESCRIPTION
**Pull request by mc-lovin**
https://github.com/square/kotlinpoet/issues/383

### CODE REVIEW

**Comment by mc-lovin** [comment]
@Egorand @swankjesse PTAL

File: src/main/java/com/squareup/kotlinpoet/TypeSpec.kt [hunk 1]
```diff
@@ -400,6 +395,9 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
     internal val funSpecs = mutableListOf<FunSpec>()
     internal val typeSpecs = mutableListOf<TypeSpec>()
     internal val isAnonymousClass get() = name == null && kind is Kind.Class
+    internal val isEnum get() = this.kind is Kind.Class && ENUM in kind.modifiers
+    internal val isAnnotation get() = this.kind is Kind.Class && ANNOTATION in kind.modifiers
+    internal val isSimpleClass = this.kind is Kind.Class && !isEnum && !isAnnotation
```
Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/TypeSpec.kt]: super nit: `this` isn't needed in these

File: src/main/java/com/squareup/kotlinpoet/TypeSpec.kt [hunk 2]
```diff
@@ -326,15 +328,8 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
...
-
-    val isAnnotation get() = this is Class && ANNOTATION in modifiers
-
-    internal val isSimpleClass get() = this is Class && !isEnum && !isAnnotation
-
-    internal val implicitPropertyModifiers get() =
-      if (isAnnotation) emptySet()
-      else defaultImplicitPropertyModifiers + when {
+    internal val implicitPropertyModifiers get() = defaultImplicitPropertyModifiers + when {
+        ANNOTATION in modifiers -> emptySet()
```
Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/TypeSpec.kt]: I feel like this isn't correct, but it's unrelated to your PR. An expect annotation should get the implicits.

Comment (Egorand) [reply src/main/java/com/squareup/kotlinpoet/TypeSpec.kt]: Guess you're right. I'll try to write a failing test and will file an issue.

**Pull request merged**